peps/pep-0812.rst

```diff
@@ -0,0 +1,651 @@
+scaling imaginary unit, i.e. ``imaginary(x) == float(x)*1j``.
+
+
+Acknowledgements
+================
+
+Thanks to Mark Dickinson for a point to the right solution and helpful
+discussion on various earlier versions of this idea.
+
+
+Footnotes
+=========
+
+.. [1] `The Fortran 2023 standard
+   <https://j3-fortran.org/doc/year/24/24-007.pdf>`_ (ISO/IEC 1539:2023)
+   §10.1.5.2.1 says:
+
+           Except for a value of type real or complex raised to an integer
+           power, if the operands have different types or kind type
+           parameters, the effect is as if each operand that differs in type
+           or kind type parameter from those of the result is converted to the
+           type and kind type parameter of the result before the operation is
+           performed.
+
+
+   Though, it's not specified now exactly operations are implemented:
+
+           Quite apart from the fact of the exclusion, the Fortran standard
+           itself contains no specification or requirement on the algorithm
+           used to calculate complex multiplication.
+
+           As was pointed out in email, there are algorithms for complex
+           multiply other than the "traditional" one.  One such algorithm
+           omits parts of the traditional calculation when the real or
+           imaginary part of one of the operands is known to be zero.
+
+           Furthermore, as the standard contains no specification or
+           requirement, it thus contains no requirement that the same
+           algorithm be used at all times.  Thus anything
+           "processor-dependent" can depend on "the phase of the moon" or
+           indeed anything else.
+
+           -- https://j3-fortran.org/doc/year/24/24-179.txt
+
+.. [2] W. Kahan and J. W. Thomas.  `Augmenting a Programming Language with
+   Complex Arithmetic
+   <http://www2.eecs.berkeley.edu/Pubs/TechRpts/1992/CSD-92-667.pdf>`_.
+   Technical Report UCB/CSD 91/667, Univ. of Calif. at Berkeley, December,
+   1991.
+
+.. [3] ISO/IEC 9899:1999, Annex G.  See `N1256 (final draft)
+   <https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf>`_.
+   https://open-std.org/, WG14. 2007.
+
+.. [4] See `Rationale for C99
+   <https://www.open-std.org/jtc1/sc22/wg14/www/C99RationaleV5.10.pdf>`_.
+   https://open-std.org/, WG14, 2003 and `Issues Regarding Imaginary Types for
+   C and C++
+   <https://wiki.edg.com/pub/CFP/WebHome/imaginary%20types%20in%20C--Thomas-Coonen.pdf>`_,
+   by Jim Thomas and Jerome T. Coonen, The Journal of C Language Translation,
+   Volume 5, Number 3, March 1994.
+
+.. [5] `Ada 2022 Reference Manual
+   <http://ada-auth.org/standards/22rm/RM-Final.pdf>`_, Annex G.
+
+.. [6] `GNU Scientific Library, Release 2.7
+   <https://www.gnu.org/software/gsl/doc/latex/gsl-ref.pdf>`_, §5.5.
+
+.. [7] `The GNU Multiple Precision Complex Library, Edition 1.3.1
+   <https://www.multiprecision.org/downloads/mpc-1.3.1.pdf>`_, §5.7.
```

Comment:
We prefer to avoid using footnotes when there's not additional info in the footnote, so we can save our readers a click to get to the important info. I think 5-7 would be better as inline links in the prose.

Reply:
I don't think they are principally different from the rest.  Important info already included in the text.  Footnotes are for additional stuff and/or references.